### PR TITLE
chore: bump tomte

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

Bumps tomte to latest version.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -36,8 +36,8 @@ def get_all_extras() -> Dict:
         "pyyaml>=4.2b1,<6.0",
         "jsonschema>=3.0.0,<4.0.0",
         "packaging>=20.3,<22.0",
-        "pytest>=7.0.0,<7.2.0",
-        "coverage>=6.4.4,<7.0.0",
+        "pytest>=7.0.0,<7.3.0",
+        "coverage>=6.4.4,<8.0.0",
```
Comment (DavidMinarsch) [inline setup.py:40]: To match latest tomte ranges

File: aea/cli/remove.py [hunk 2]
```diff
@@ -372,7 +372,7 @@ def remove(self) -> None:
     @property
     def agent_items(self) -> Set[PublicId]:
         """Return items registered with agent of the same type as item."""
-        return getattr(self.agent_config, self.item_type_plural, set)
+        return getattr(self.agent_config, self.item_type_plural, set())
```
Comment (DavidMinarsch) [inline aea/cli/remove.py:375]: found a bug

File: benchmark/checks/check_multiagent.py [hunk 3]
```diff
@@ -158,8 +158,8 @@ def run(
     runner = AEARunner(agents, runner_mode)
     runner.start(threaded=True)
 
-    for agent in agents:
-        wait_for_condition(lambda: agent.is_running, timeout=5)
+    for agent_ in agents:
+        wait_for_condition(lambda a=agent_: a.is_running, timeout=5)
```
Comment (DavidMinarsch) [inline benchmark/checks/check_multiagent.py:162]: this was a bug!

File: benchmark/checks/check_multiagent_http_dialogues.py [hunk 4]
```diff
@@ -185,8 +185,8 @@ def run(
 
     runner = AEARunner(agents, runner_mode)
     runner.start(threaded=True)
-    for agent in agents:
-        wait_for_condition(lambda: agent.is_running, timeout=5)
+    for agent_ in agents:
+        wait_for_condition(lambda a=agent_: a.is_running, timeout=5)
```
Comment (DavidMinarsch) [inline benchmark/checks/check_multiagent_http_dialogues.py:189]: this too!

File: plugins/aea-cli-benchmark/aea_cli_benchmark/case_multiagent/case.py [hunk 5]
```diff
@@ -146,8 +146,8 @@ def run(
     runner = AEARunner(agents, runner_mode)
     runner.start(threaded=True)
 
-    for agent in agents:
-        wait_for_condition(lambda: agent.is_running, timeout=5)
+    for agent_ in agents:
+        wait_for_condition(lambda a=agent_: a.is_running, timeout=5)
```
Comment (DavidMinarsch) [inline plugins/aea-cli-benchmark/aea_cli_benchmark/case_multiagent/case.py:150]: !

File: plugins/aea-cli-benchmark/aea_cli_benchmark/case_multiagent_http_dialogues/case.py [hunk 6]
```diff
@@ -170,8 +170,8 @@ def run(
 
     runner = AEARunner(agents, runner_mode)
     runner.start(threaded=True)
-    for agent in agents:
-        wait_for_condition(lambda: agent.is_running, timeout=5)
+    for agent_ in agents:
+        wait_for_condition(lambda a=agent_: a.is_running, timeout=5)
```
Comment (DavidMinarsch) [inline plugins/aea-cli-benchmark/aea_cli_benchmark/case_multiagent_http_dialogues/case.py:174]: !

File: setup.cfg [hunk 7]
```diff
@@ -13,7 +13,7 @@ exclude=.md,
     scripts/oef/launch.py
 max-line-length = 88
 select = B,C,D,E,F,I,W,
-ignore = E203,E501,W503,D202,B014,D400,D401,DAR
+ignore = E203,E501,W503,D202,B014,D400,D401,DAR,B028,B017
```
Comment (DavidMinarsch) [inline setup.cfg:16]: Need to enable at some point

File: aea/launcher.py [hunk 8]
```diff
@@ -173,7 +173,7 @@ def create_async_task(self, loop: AbstractEventLoop) -> TaskAwaitable:
             raise ValueError(
                 "Agent runtime is not async compatible. Please use runtime_mode=async"
             )
-        return loop.create_task(self._agent.runtime.start_and_wait_completed())
+        return loop.create_task(self._agent.runtime.start_and_wait_completed())  # type: ignore
```
Comment (DavidMinarsch) [inline aea/launcher.py:176]: @solarw for you to investigate. I just added the ignore temporarily. It is masking an issue!

File: aea/runner.py [hunk 9]
```diff
@@ -79,7 +79,7 @@ def create_async_task(self, loop: AbstractEventLoop) -> TaskAwaitable:
             raise ValueError(
                 "Agent runtime is not async compatible. Please use runtime_mode=async"
             )
-        return loop.create_task(self._agent.runtime.start_and_wait_completed())
+        return loop.create_task(self._agent.runtime.start_and_wait_completed())  # type: ignore
```
Comment (DavidMinarsch) [inline aea/runner.py:82]: @solarw for you to investigate. I just added the ignore temporarily. It is masking an issue!

File: scripts/check_ipfs_hashes_pushed.py [hunk 10]
```diff
@@ -31,18 +32,28 @@
 
 IPFS_ENDPOINT = "https://gateway.autonolas.tech/ipfs"
 MAX_WORKERS = 10
-REQUEST_TIMEOUT = 5  # seconds
+REQUEST_TIMEOUT = 10  # seconds
 
 
-def check_ipfs_hash_pushed(ipfs_hash: str) -> Tuple[str, bool]:
+def check_ipfs_hash_pushed(ipfs_hash: str, retries: int = 5) -> Tuple[str, bool]:
```
Comment (DavidMinarsch) [inline scripts/check_ipfs_hashes_pushed.py:38]: @8ball030 this should make you happy ;)

File: packages/valory/connections/ledger/tests/conftest.py [hunk 11]
```diff
@@ -126,7 +127,7 @@ def make_ledger_api_connection(
     return connection
 
 
-@pytest.fixture()
```
Comment (DavidMinarsch) [inline packages/valory/connections/ledger/tests/conftest.py:129]: breaking change in this library

File: .github/workflows/workflow.yml [hunk 12]
```diff
@@ -58,7 +58,7 @@ jobs:
           sudo apt-get update --fix-missing
           sudo apt-get autoremove
           sudo apt-get autoclean
-          pip install tomte[tox]==0.1.5
+          pip install tomte[tox]==0.2.2
```
Comment (solarw) [inline .github/workflows/workflow.yml:61]: make ci-requirements.txt and use `pip install -r` to make version edit easier?

Comment (DavidMinarsch) [reply .github/workflows/workflow.yml:61]: Good idea. Will open a Trello ticket

**Review by solarw — APPROVED** [review]
LGTM

**Review by angrybayblade — APPROVED** [review]

File: aea/cli/fetch.py [hunk 13]
```diff
@@ -20,7 +20,7 @@
 """Implementation of the 'aea fetch' subcommand."""
 import os
 import shutil
-from distutils.dir_util import copy_tree
+from distutils.dir_util import copy_tree  # pylint: disable=deprecated-module
```
[the remark this answers is not in the document]
Comment (DavidMinarsch) [reply aea/cli/fetch.py:23]: Probably

File: benchmark/checks/check_multiagent.py (hunk 3 above)
Comment (solarw) [reply benchmark/checks/check_multiagent.py:162]: actually not a bug, it's used inplace. so closure is not needed.
but automatic check can suppose closure is needed.

File: benchmark/checks/check_multiagent_http_dialogues.py (hunk 4 above)
Comment (solarw) [reply benchmark/checks/check_multiagent_http_dialogues.py:189]: [same here](https://github.com/valory-xyz/open-aea/pull/543#discussion_r1081179738)

File: plugins/aea-cli-benchmark/aea_cli_benchmark/case_multiagent/case.py (hunk 5 above)
Comment (solarw) [reply plugins/aea-cli-benchmark/aea_cli_benchmark/case_multiagent/case.py:150]: [same here](https://github.com/valory-xyz/open-aea/pull/543#discussion_r1081179738)

File: plugins/aea-cli-benchmark/aea_cli_benchmark/case_multiagent_http_dialogues/case.py (hunk 6 above)
Comment (solarw) [reply plugins/aea-cli-benchmark/aea_cli_benchmark/case_multiagent_http_dialogues/case.py:174]: [same here](https://github.com/valory-xyz/open-aea/pull/543#discussion_r1081179738)

**Pull request merged**